crates/ruff_linter/src/rules/pylint/rules/bad_dunder_method_name.rs

```diff
@@ -45,13 +51,19 @@ use crate::rules::pylint::helpers::is_known_dunder_method;
 #[violation]
 pub struct BadDunderMethodName {
     name: String,
+    kind: Kind,
 }
 
 impl Violation for BadDunderMethodName {
     #[derive_message_formats]
     fn message(&self) -> String {
-        let BadDunderMethodName { name } = self;
-        format!("Bad or misspelled dunder method name `{name}`")
+        let BadDunderMethodName { name, kind } = self;
+        match kind {
+            Kind::Misspelled => format!("Bad or misspelled dunder method name `{name}`"),
+            Kind::RemovedInPython3 => {
+                format!("Python 2 or older only dunder method name `{name}`")
```

Comment:
@AlexWaygood any suggestion on how to improve the wording here? It's kind of hard to read

Reply:
I think removing the 'only' makes it read a bit smoother, but it might change the meaning slightly.

Reply:
Maybe something like this?

```suggestion
                format!("Dunder method `{name}` has no special meaning in Python 3")
```

Though actually, that's a pretty good description of what the problem is in both cases. I wonder if we need to distinguish between misspelled dunder methods and those that were removed in Python 3 at all, or if we could just improve the docs (like this PR already does) and use a better error message here that applies to both cases?

Reply:
Way clearer!

Reply:
@MichaReiser What do you think about a single message for both cases?

Reply:
No problem, I also agree 😄